pkg/controller/common/helpers.go

```diff
@@ -503,6 +506,101 @@ func ParseAndConvertConfig(rawIgn []byte) (ign3types.Config, error) {
 	}
 }
 
+// Internal errors used for base64-decoding and gunzipping Ignition configs
+var (
+	errConfigNotGzipped       = fmt.Errorf("ignition config not gzipped")
+	errConfigNotBase64Encoded = fmt.Errorf("ignition config not base64 encoded")
+)
+
+// Decode, decompress, and deserialize an Ignition config file.
+func ParseAndConvertGzippedConfig(rawIgn []byte) (ign3types.Config, error) {
+	// Let's try to decode and decompress our payload
+	out, err := decodeAndDecompressPayload(bytes.NewReader(rawIgn))
+	if err == nil {
+		// Our payload was decoded and decompressed, so let's deserialize it.
+		return ParseAndConvertConfig(out)
+	}
+
+	// Our Ignition config is not base64-encoded, which means it might just be gzipped.
+	if errors.Is(err, errConfigNotBase64Encoded) {
+		out, err = decompressPayload(bytes.NewReader(rawIgn))
+		glog.V(2).Info("ignition config was not base64 encoded, trying to gunzip ignition config")
```

Comment:
I think you meant this to go above the decompress function?

Reply:
I've done a bit of refactoring so it might make sense now.

Reply:
V(2) is the default verbosity I believe. I think it's probably fine as is for now, but maybe something to consider in terms of verbosity in the future